docs/debugging-python-lambda-functions.md

```diff
@@ -89,4 +86,65 @@ python -m pip install -r requirements.txt
     echo '{}' | sam local invoke HelloWorldFunction -d 5678
     ```
 
-3. When you see `waiting for debugger to attach...`, go back to Visual Studio Code and press F5.
+3. When you see `waiting for debugger to attach...`, go back to Visual Studio Code and press F5 to attach the debugger to the handler that you invoked in the previous step.
+
+## Optional: Automatically start debugging when ready
+
+With the above steps, you need to manually invoke SAM CLI from the command line, wait for it to be ready, then attach the debugger. We can automate the process of invoking SAM CLI and waiting for it to be ready by using a `preLaunchTask`.
+
+1. Open `<app root>/.vscode/tasks.json` (create a new file if it does not already exist).
+2. Add the following contents to `tasks.json`:
+
+    ```json
+    {
+        "label": "Debug Python Lambda Function",
+        "type": "shell",
+        "command": "sam",
+        "args": [
+            "local",
+            "invoke",
+            "HelloWorldFunction", // Replace this with the resource name of your lambda function from your Serverless Application template.yaml file
+            "--template",
+            "${workspaceFolder}/template.yaml", // Replace this with the appropriate workspace-relative path to your Serverless Application template.yaml file
+            "--event",
+            "${workspaceFolder}/event.json", // Replace this with the appropriate workspace-relative path to your event.json file
+            "-d",
+            "5678"
+        ],
+        "isBackground": true,
+        "problemMatcher": {
+            "pattern": [
+                {
+                    // Use regex that never matches anything.
+                    "regexp": "^(x)(\\b)(x)$",
+                    "file": 1,
+                    "location": 2,
+                    "message": 3
+                }
+            ],
+            "background": {
+                // This is how the debugger knows when it can attach
+                "activeOnStart": true,
+                "beginsPattern": "^Fetching lambci.* Docker container image......$",
+                "endsPattern": "^waiting for debugger to attach...$"
+            }
+        }
+    }
+    ```
+
+3. Open `<app root>/.vscode/launch.json`, and add the following property to the configuration that you created earlier:
+
+    ```json
+    {
+        "version": "0.2.0",
+        "configurations": [
+            {
```

Comment:
Does this block need any other lines? I wanted to check because it seems tiny, and doesn't include something like type: "launch"

Reply:
These instructions are for modifying the config that you created earlier--the `// ...` is intended to represent the omitted lines. I'll re-work this to make it more clear.